File: Java_Virtual_World/src/Gameplay/Animals/Antelope.java
package Gameplay.Animals;

import Gameplay.Organism;
import java.lang.Math;

public class Antelope extends Animal {
    public static final int STRENGTH = 4;
    public static final int INITIATIVE = 4;
    public static final int RANGE = 2;
    public static final double PROBABILITY = 0.5;

    public Antelope(int x, int y){
        super(x ,y , STRENGTH, INITIATIVE);
    }

    @Override
    public void Action(){
        randomMove(RANGE);
    }
    @Override
    public boolean didEscape(){
        return Math.random() < PROBABILITY;
    }
    @Override
    public String Draw(){
        return "🦌";
    }
    @Override
    public Organism Copy(){
        return new Antelope(x,y);
    }
    @Override
    public String toString(){
        return "Antelope";
    }
}

File: Python_Virtual_World/Antelope.py
from random import random

from MyVector import MyVector
from Animal import Animal


class Antelope(Animal):


    STRENGTH = 4
    INITIATIVE = 4
    RANGE = 2
    PROPABILITY = 0.5


    def __init__(self, position : MyVector):
        super().__init__(position, Antelope.STRENGTH, Antelope.INITIATIVE)


    def action(self):

        self._RandomMove(Antelope.RANGE)

    def didEscape(self) -> bool:

        if random() < Antelope.PROPABILITY:
            self._world.getNots().add(self.__str__() + " has escaped!")


    def draw(self) -> str:

        return "#964B00"


    def __str__(self):

        return "Antelope"